UserAgentBag.js

```diff
@@ -93,6 +52,72 @@ class UserAgentBag {
 
 module.exports = UserAgentBag
 
+function parse (arg) {
+  if (arg == null) {
+    return {
+      nodes: [],
+      asMap: new Map()
+    }
+  } else if (typeof arg === 'string') {
+    return parseString(arg)
+  } else if (arg && arg[Symbol.iterator]) {
+    return parseIterable(arg)
+  } else {
+    throw new TypeError('UserAgentBag must be constructed with a string or an iterable')
+  }
+}
+
+function parseString (str) {
+  let nodes = []
+  const asMap = new Map()
+
+  if (str.length && (str.length <= MAX_STRING_LENGTH)) {
+    try {
+      nodes = parser.parse(str)
+    } catch (err) {}
```

Comment:
i'm OK with returning an empty bag when the string is invalid (although would prefer something more explicit), but how do you feel about either adding
```js
get size() {
  return this._asMap.size;
}
```
or
```js
get isEmpty() {
  return this._asMap.size === 0;
}
```

to make checking for that case slighly easier?

Reply:
I like the `size` method, and I think we should add that in a followup. Created #2.